app/models/dog.rb

```diff
@@ -143,19 +153,40 @@ class Dog < ApplicationRecord
 
   before_save :update_adoption_date
 
-  scope :is_age,                      ->(age) { where age: age }
-  scope :is_size,                     ->(size) { where size: size }
-  scope :is_status,                   ->(status) { where status: status }
-  scope :cb_high_priority,            ->(_) { where is_high_priority: true }
-  scope :cb_medical_need,             ->(_) { where has_medical_need: true }
-  scope :cb_medical_review_needed,    ->(_) { where medical_review_complete: false }
-  scope :cb_special_needs,            ->(_) { where is_special_needs: true }
-  scope :cb_behavior_problems,        ->(_) { where has_behavior_problem: true }
-  scope :cb_foster_needed,            ->(_) { where needs_foster: true }
-  scope :cb_spay_neuter_needed,       ->(_) { where is_altered: false }
-  scope :cb_no_cats,                  ->(_) { where no_cats: true }
-  scope :cb_no_dogs,                  ->(_) { where no_dogs: true }
-  scope :cb_no_kids,                  ->(_) { where no_kids: true }
+  scope :is_age,                                  ->(age) { where age: age }
+  scope :is_size,                                 ->(size) { where size: size }
+  scope :is_status,                               ->(status) { where status: status }
+  scope :is_breed,                                ->(breed_partial) { joins("join breeds on (breeds.id = dogs.primary_breed_id) or (breeds.id = dogs.secondary_breed_id)").where("breeds.name ilike '%#{sanitize_sql_like(breed_partial)}%'").distinct }
```

Comment:
hakiri flags possible sql injection vulnerability for line 159, in spite of the fact that the user-submitted text is sanitized with the sanitize_sql_like method. Also, we know it's admin user submitting this form, not random bad person on the internet.